benches/bench.rs

```diff
@@ -0,0 +1,48 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use prima_datadog::{
+    configuration::{Country, PrimaConfiguration},
+    Datadog,
+};
+
+fn incr_benchmark(c: &mut Criterion) {
+    let configuration = PrimaConfiguration::new(
+        "0.0.0.0:1234",
+        "0.0.0.0:0",
+        "prima_datadog_benchmarks",
+        "dev".parse().unwrap(),
+    )
+    .with_country(Country::It);
+    // 20 test tags to simulate a normal to heavily tagged metric
+    let tags = vec![
+        "1".to_string(),
+        "2".to_string(),
+        "3".to_string(),
+        "4".to_string(),
+        "5".to_string(),
+        "6".to_string(),
+        "7".to_string(),
+        "8".to_string(),
+        "9".to_string(),
+        "10".to_string(),
+        "11".to_string(),
+        "12".to_string(),
+        "13".to_string(),
+        "14".to_string(),
+        "15".to_string(),
+        "16".to_string(),
+        "17".to_string(),
+        "18".to_string(),
+        "19".to_string(),
+        "20".to_string(),
+    ];
+    Datadog::init(configuration).unwrap();
+    c.bench_function("incr_benchmark", |b| {
+        b.iter(|| {
+            // Note: clone here is ok since we only care about /relative/ perf here
+            Datadog::incr("test", tags.clone());
+        });
+    });
+}
+
```

Comment:
I think we should have a benchmark for the unhappy path where we've exceeded the tag cardinality limit

Reply:
Added. Results relative to baseline (well, to track_tags being a no-op) below. Regression is basically the same in both cases, paying about 0.4us for the tracking, when there are about 20 tags
```
incr_benchmark          time:   [3.2138 µs 3.2190 µs 3.2249 µs]
                        change: [+12.648% +14.018% +15.098%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 14 outliers among 100 measurements (14.00%)
  1 (1.00%) low mild
  6 (6.00%) high mild
  7 (7.00%) high severe

incr_with_too_many_tags time:   [3.5162 µs 3.5211 µs 3.5272 µs]
                        change: [+14.942% +15.783% +16.503%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 9 outliers among 100 measurements (9.00%)
  3 (3.00%) low mild
  4 (4.00%) high mild
  2 (2.00%) high severe
```

Reply:
I think it's better to make tests that assert behavior (like that the event is sent once and only once), the bench is only of convenience. The test included in this PR checks the scenario you describe. Tests run in CI and are pass fail, benches are more of a dev convenience while working to make sure you didn't mess up (in my opinion)